### PR TITLE
Popup

### PR DESCRIPTION
**Pull request by chibDevika**

### CODE REVIEW

File: src/Components/Calendar.jsx [hunk 1]
```diff
@@ -2,16 +2,35 @@ import React, { useState, useEffect, useCallback } from 'react';
...
   const [myEvents, setMyEvents] = useState([]);
+  const [open, setOpen] = React.useState(false);
+  const [buttonClick, setButtonClick] = useState(false);
+  const [responseText, setResponseText] = useState('');
+  const [selected, setSelected] = useState();
+  const handleClose = () => {
+    setButtonClick(false);
+    setResponseText('');
+    setOpen(false);
+  };
```
Comment (shubham-gup) [inline src/Components/Calendar.jsx:29]: Should this be memoized?

File: src/Components/Calendar.jsx [hunk 2]
```diff
@@ -76,19 +94,121 @@ function MyCalendar() {
...
   }, [getEvents, logout]);
 
+  const handleSelected = (event) => {
+    setSelected(event);
+    setStartDate(event.start);
+    setEndDate(event.end);
+    SetLeaveID(event.id);
+    setReason(event.title.split('Reason:')[1]);
+    setOpen(true);
+  };
```
Comment (shubham-gup) [inline src/Components/Calendar.jsx:104]: Should this be memoized? Also, why is setter starting with a capital letter? `SetLeaveID`?

File: src/Components/Calendar.jsx [hunk 3]
```diff
@@ -76,19 +94,121 @@ function MyCalendar() {
...
+                </Typography>
+                <TextField
+                  id="outlined-basic"
+                  label="Mention your reason"
+                  variant="outlined"
+                  multiline
+                  defaultValue={reason}
+                  onChange={(input) => {
+                    setReason(input.target.value);
+                  }}
```
Comment (shubham-gup) [inline src/Components/Calendar.jsx:177]: Should we pass a method here? instead of inline?

File: src/Components/Calendar.jsx [hunk 4]
```diff
@@ -76,19 +94,121 @@ function MyCalendar() {
...
+                  <Typography margin="auto">{responseText}</Typography>
+                ) : null}
+              </Box>
+              <Button
+                onClick={() =>
+                  saveChange(
+                    moment(start).format('YYYY-MM-DD HH:mm:ss'),
+                    moment(end).format('YYYY-MM-DD HH:mm:ss'),
+                    { reason }.reason,
+                  )}
```
Comment (shubham-gup) [inline src/Components/Calendar.jsx:193]: Should we pass a method here? instead of inline?

**Review by shubham-gup — CHANGES_REQUESTED** [review]

File: src/Components/Calendar.jsx (hunk 1 above)
Comment (chibDevika) [reply src/Components/Calendar.jsx:29]: done

File: src/Components/Calendar.jsx (hunk 2 above)
Comment (chibDevika) [reply src/Components/Calendar.jsx:104]: corrected that

**Pull request merged**